### ISSUE TITLE
[Search a11y] aria-hidden property is set to true on focusable search elements

### ISSUE THREAD

**Issue by CassidyJensen** [carbon-design-system/carbon#7358]
## Environment

> Operating system

> Browser

> Automated testing tool and ruleset

> Assistive technology used to verify

## Detailed description

> What version of the Carbon Design System are you using?
Carbon-components-react

> What did you expect to happen?
aria-hidden should be false for focusable elements. 

> What happened instead?
aria-hidden is set to true.

> What WCAG 2.1 checkpoint does the issue violate?
Rule ID: aria-hidden-focus
Ruleset: axe-core 3.4
User Impact: Serious 
WCAG: 4.1.2

## Steps to reproduce the issue

1. Find a persistent data table search
2. Inspect the search input element.
3. Check value for aria-hidden. 

[https://codesandbox.io/s/datatable-with-batch-actions-uolbz](https://codesandbox.io/s/datatable-with-batch-actions-uolbz
)
## Additional information

Using the aria-hidden="true" attribute on an element removes the element and ALL of its child nodes from the accessibility API making it completely inaccessible to screen readers and other assistive technologies. Aria-hidden may be used with extreme caution to hide visibly rendered content from assistive technologies only if the act of hiding this content is intended to improve the experience for users of assistive technologies by removing redundant or extraneous content. If aria-hidden is used to hide visible content from screen readers, the identical or equivalent meaning and functionality must be exposed to assistive technologies.

aria-hidden should not be true for items that allow input or focus.

**Comment by dakahn** [comment]
`aria-hidden` is set on the search icon, not the element itself. Without search results the element is represented in the DOM with a search landmark like this
![image](https://user-images.githubusercontent.com/40970507/100947346-188f9300-34cb-11eb-8bec-3e6f513e0db6.png)

and in the Accessibility Tree like this
![image](https://user-images.githubusercontent.com/40970507/100947453-4bd22200-34cb-11eb-9bcb-7f02f5ddd571.png)

When testing with NVDA the search element worked and spoke ok. That said the fancy search implementation (reveals itself visually only on focus...) is problematic -- and it's labelings could be more descriptive than just "filter results".

**Comment by CassidyJensen** [comment]
The icon does have an aria-hidden, which is fine. The issue is that it is also present on the input itself: 
![image](https://user-images.githubusercontent.com/45407808/101192765-c9f00f00-3618-11eb-98b2-0f28e21d464b.png)
